server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -1226,6 +1226,20 @@ func (s *BuildBuddyServer) serveArtifact(ctx context.Context, w http.ResponseWri
 			log.Warningf("Error serving invocation-%s.log: %s", iid, err)
 			http.Error(w, "Internal server error", http.StatusInternalServerError)
 		}
+	case "timing_profile":
+		name := params.Get("name")
+		b, err := s.env.GetBlobstore().ReadBlob(ctx, iid+"/artifacts/timing_profile/"+name)
```

Comment:
I think buffering the blob in memory is fine for now since these are compressed, but we might want to consider adding a blobstore.Reader method in the future. Might be worth adding prometheus metrics for blobstore Read/Write so we can keep an eye on this

Reply:
Yeah, I was thinking we probably should add a Reader; hopefully it will be less problematic than the Writer :P
I'll add one before submitting the test logs PR.